### ISSUE TITLE
Refactor search service's searchInstructor method

### ISSUE THREAD

**Issue by t-cheepeng** [TEAMMATES/teammates#10212]
**Details**
Due to the fixing of student tables not responding to changes(#10196) and changing model type for `student-list` component, the `StudentListSectionData` is going to be redundant. Refactor out the search service to use the new `StudentListRowModel` instead

**Comment by edwardkang0925** [comment]
Hi,
My partner and I would like to work on this issue if possible.

**Comment by t-cheepeng** [comment]
@edwardkang0925 Hi, your team is welcomed to work on this issue. Do seek help if you run into any issues

**Comment by edwardkang0925** [comment]
@t-cheepeng  Thank you so much for offering help.
Actually, I have a couple of questions to ask so that I am not doing what I am not supposed to do for this issue. 

My original approach was to replace every SearchStudentsTable with SearchStudentsListRowTable because SearchStudentsTable includes StudentListSectionData.
In order to do so, I changed InstructorSearchResult to have the SearchStudentsListRowTable instead of the SearchStudentsTables and started to change getCoursesWithSections, getPrivileges, and combinePrivileges functions to support the new data type (SearchStudentsListRowTable). 
By proceeding as mentioned above, I deleted the flattenStudentTable function you wrote in InstructorSearchPageComponent, and now it seems I have to fix the search.service.ts file because the data type for InstructorSearchResult and return types of the three functions I have modified are changed. 

Before I proceed any further fixing the component tests, I would appreciate some advice on whether I am following the best practice or straying from the right track. 

In case you want to see my changes, here is the link.
https://github.com/TEAMMATES/teammates/compare/master...edwardkang0925:RefactorSearch
